python/rikai/parquet/dataset.py

```diff
@@ -228,3 +225,50 @@ def __iter__(self):
             yield self._convert(
                 shuffler.pop().to_dict(), self.spark_row_metadata
             )
+
+    def to_pandas(self):
+        filesystem, path = fs.FileSystem.from_uri(self.uri)
+        raw_df = (
+            pq.ParquetDataset(path, filesystem).read(self.columns).to_pandas()
+        )
+        types = {
+            f["name"]: f["type"] for f in self.spark_row_metadata["fields"]
+        }
+        return pd.DataFrame(
+            {
+                name: self._convert_col(col, types.get(name))
+                for name, col in raw_df.iteritems()
+            }
+        )
+
+    def _convert_col(self, col: pd.Series, field_type) -> pd.Series:
+        if field_type is None:
+            return col
+        if not isinstance(field_type, dict):
+            return col
+        elif field_type["type"] == "udt":
+            udt = self._find_udt(field_type["pyClass"])
+            return col.apply(partial(_convert_udt_value, udt=udt))
+        elif field_type["type"] == "struct":
+            return col.apply(lambda d: self._convert(d, field_type))
+        elif field_type["type"] == "array":
+
+            def convert_array(arr):
```

Comment:
shall we move this to be a standalone function globally?

Reply:
yeah, good idea this is usable for the self._convert as well